x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/amazonbedrock/client/AmazonBedrockStreamingProcessor.java

```diff
@@ -67,33 +47,23 @@ public void onSubscribe(Flow.Subscription subscription) {
         }
     }
 
-    @Override
-    public void onNext(ConverseStreamOutput item) {
-        if (item.sdkEventType() == ConverseStreamOutput.EventType.CONTENT_BLOCK_DELTA) {
-            demand.set(0); // reset demand before we fork to another thread
-            item.accept(ConverseStreamResponseHandler.Visitor.builder().onContentBlockDelta(this::sendDownstreamOnAnotherThread).build());
+    public void subscribe(Flow.Subscriber<? super T> subscriber) {
+        if (downstream == null) {
+            downstream = subscriber;
+            downstream.onSubscribe(new StreamSubscription());
         } else {
-            upstream.request(1);
+            subscriber.onError(new IllegalStateException("Subscriber already set."));
         }
     }
 
-    // this is always called from a netty thread maintained by the AWS SDK, we'll move it to our thread to process the response
-    private void sendDownstreamOnAnotherThread(ContentBlockDeltaEvent event) {
-        runOnUtilityThreadPool(() -> {
-            var text = event.delta().text();
-            var result = new ArrayDeque<StreamingChatCompletionResults.Result>(1);
-            result.offer(new StreamingChatCompletionResults.Result(text));
-            var results = new StreamingChatCompletionResults.Results(result);
-            downstream.onNext(results);
-        });
-    }
-
-    @Override
     public void onError(Throwable amazonBedrockRuntimeException) {
         ExceptionsHelper.maybeDieOnAnotherThread(amazonBedrockRuntimeException);
         error.set(
             new ElasticsearchException(
-                Strings.format("AmazonBedrock StreamingChatProcessor failure: [%s]", amazonBedrockRuntimeException.getMessage()),
+                org.elasticsearch.core.Strings.format(
```

Comment:
nit: Do we need the full path here? Can the import just be `Strings.format()` like it was before?